nisystemlink/clients/testmonitor/_test_monitor_client.py

```diff
@@ -26,6 +39,159 @@ def api_info(self) -> models.ApiInfo:
             Information about available API operations.
 
         Raises:
-            ApiException: if unable to communicate with the `nitestmonitor` service.
+            ApiException: if unable to communicate with the `ni``/nitestmonitor``` service.
+        """
+        ...
+
+    @post("products", args=[Field("products")])
+    def create_products(
+        self, products: List[Product]
+    ) -> models.CreateProductsPartialSuccess:
+        """Creates one or more products and returns errors for failed creations.
+
+        Args:
+            products: A list of products to attempt to create.
+
+        Returns: A list of created products, products that failed to create, and errors for
+            failures.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` service of provided invalid
+                arguments.
+        """
+        ...
+
+    @get(
+        "products",
+        args=[Query("continuationToken"), Query("take"), Query("returnCount")],
+    )
+    def get_products(
+        self,
+        continuation_token: Optional[str] = None,
+        take: Optional[int] = None,
+        return_count: Optional[bool] = None,
+    ) -> models.PagedProducts:
+        """Reads a list of products.
+
+        Args:
+            continuation_token: The token used to paginate results.
+            take: The number of products to get in this request.
+            return_count: Whether or not to return the total number of products available.
+
+        Returns:
+            A list of products.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @get("products/{id}")
+    def get_product(self, id: str) -> models.Product:
+        """Retrieves a single product by id.
+
+        Args:
+            id (str): Unique ID of a products.
+
+        Returns:
+            The single product matching `id`
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @post("query-products")
+    def query_products(
+        self, query: models.QueryProductsRequest
+    ) -> models.PagedProducts:
```

Comment:
I'd like to get this in soon so that it's not hanging over me still. What is the timeline for @santhoshramaraj getting that Generator change in?

Reply:
@adamarnesen As this is the first release of `ProductClient` it is good to make the generator concept as the default implementation for paged responses, please refer to #75 for the concept for `query_tables`

Reply:
@santhoshramaraj @mure it looks like in #75 that the generator pattern is just an addon that _uses_ the Paged based method. Therefore, adding generator will not break this API. I think that I'd prefer to add the generator in a separate PR after we have approved naming conventions in #75. It will make the changes clearer.

Reply:
@adamarnesen, @mure and I were debating whether to make the generator implementation as the default one instead of an addon, so, `query_tables` could be generator and `_query_tables` or `query_tables_paged` could be the underlying paged implementation.

Reply:
@mure  I need some guidance from you on this one. If we want to implement generator pattern, I'd prefer to change the name of the paged functions in this PR and then do a follow-on with the generator addition. That would keep compatibility and let us move forward on this change.

Reply:
My vote is for `query_products_paged`

Reply:
I've made this change. Closing this thread.

Reply:
I'm in support of this approach 👍